### PR TITLE
OpenCV

### PR DESCRIPTION
**Pull request by lowerkase**
Uses OpenCV to determine how many rings there are and drive to the corresponding square on field

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ConceptVuMarkIdentification.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
+/* Copyright (c) 2017 FIRST. All rights reserved.
```
Comment (je3m) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ConceptVuMarkIdentification.java]: this file looks like it's vuforia. If we are doing OpenCV instead,

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ConceptVuforiaUltimateGoalNavigation.java [hunk 2]
```diff
@@ -0,0 +1,301 @@
+/* Copyright (c) 2019 FIRST. All rights reserved.
```
Comment (je3m) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ConceptVuforiaUltimateGoalNavigation.java]: this file is also code that is not part of the robot

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java [hunk 3]
```diff
@@ -16,10 +16,27 @@
     Telemetry telemetry;
     HardwareInnov8Hera hera;
     LinearOpMode opMode;
-
-    public static double INCH_TO_TICK = (360/6); // The number of encoder ticks per inch for our wheels
-    public static double SIDE_INCH_TO_TICK = (360/6); // The number of encoder ticks for one inch while travelling sideways, change later
-
+    private double wheelPower = 1;
```
Comment (je3m) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java:19]: Did you actually change any drive code? this looks like a merge error

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/HardwareInnov8Hera.java [hunk 4]
```diff
@@ -32,10 +33,10 @@
     public DcMotor motorTwo = null; // Back left  wheel
     public DcMotor motorThree = null; // Front right wheel
     public DcMotor motorFour = null; // Back right wheel
-    public DcMotor shooterMotor = null;
+    //public DcMotor shooterMotor = null;
```
Comment (je3m) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/HardwareInnov8Hera.java]: commented code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/HardwareInnov8Hera.java [hunk 5]
```diff
@@ -59,36 +61,43 @@ public void init(HardwareMap ahwMap) {
...
-        shooterMotor = this.hwMap.get(DcMotor.class, "shooterMotor");
+        //shooterMotor = this.hwMap.get(DcMotor.class, "shooterMotor");
 
         // Using REV (I think) motors
         motorOne.setDirection(DcMotor.Direction.FORWARD);
         motorTwo.setDirection(DcMotor.Direction.REVERSE);
         motorThree.setDirection(DcMotor.Direction.FORWARD);
         motorFour.setDirection(DcMotor.Direction.REVERSE);
-        shooterMotor.setDirection(DcMotor.Direction.FORWARD);
+        //shooterMotor.setDirection(DcMotor.Direction.FORWARD);
```
Comment (je3m) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/HardwareInnov8Hera.java]: commented code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/HardwareInnov8Hera.java [hunk 6]
```diff
@@ -59,36 +61,43 @@ public void init(HardwareMap ahwMap) {
         motorTwo = this.hwMap.get(DcMotor.class, "motorTwo");
         motorThree = this.hwMap.get(DcMotor.class, "motorThree");
         motorFour = this.hwMap.get(DcMotor.class, "motorFour");
-        shooterMotor = this.hwMap.get(DcMotor.class, "shooterMotor");
+        //shooterMotor = this.hwMap.get(DcMotor.class, "shooterMotor");
```
Comment (je3m) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/HardwareInnov8Hera.java]: commented code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/HardwareInnov8Hera.java [hunk 7]
```diff
@@ -59,36 +61,43 @@ public void init(HardwareMap ahwMap) {
...
-        parameters.loggingTag = "IMU";
-        parameters.accelerationIntegrationAlgorithm = new JustLoggingAccelerationIntegrator();
 
         // Set all motors to zero power
         motorOne.setPower(0);
         motorTwo.setPower(0);
         motorThree.setPower(0);
         motorFour.setPower(0);
-        shooterMotor.setPower(0);
+        //shooterMotor.setPower(0);
```
Comment (je3m) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/HardwareInnov8Hera.java]: commented code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/HardwareInnov8Hera.java [hunk 8]
```diff
@@ -32,10 +33,10 @@
     public DcMotor motorTwo = null; // Back left  wheel
     public DcMotor motorThree = null; // Front right wheel
     public DcMotor motorFour = null; // Back right wheel
-    public DcMotor shooterMotor = null;
+    //public DcMotor shooterMotor = null;
 
     public BNO055IMU imu;
-
+    BNO055IMU.Parameters parameters;
```
Comment (je3m) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/HardwareInnov8Hera.java:38]: What is this instance variable for?
Does it need to be an instance variable?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Hera.java [hunk 9]
```diff
@@ -40,9 +48,54 @@ public void teleop(Gamepad gamepad1, Gamepad gamepad2) {
     }
     
 
-    public void forwardTwelveInches() {
-        
-        driveTrain.goForward(12);
+    public void forwardTurn() {
```
Comment (je3m) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Hera.java]: this function does not do what its name says

**Review by MayaExists — APPROVED** [review]

**Pull request merged**